macros/src/delegates.rs

```diff
@@ -12,7 +12,7 @@ type AsyncData = BoxFuture<Value, Error>;
 
 struct DelegateMethod<T, F> {
 	delegate: Arc<T>,
-	closure: F,
+	closure: Arc<F>,
```

Comment:
Unfortunatelly `delegate` is shared between many methods, so `Arc<T>` is required anyway.